server/src/main/java/org/elasticsearch/cluster/service/ClusterApplierService.java

```diff
@@ -274,6 +274,10 @@ public void run() {
                     if (timeout != null) {
                         notifyTimeout.cancellable = threadPool.schedule(notifyTimeout, timeout, ThreadPool.Names.GENERIC);
                     }
+                    if (lifecycle.stoppedOrClosed()) {
```

Comment:
Could we not do this 3 lines earlier? No need to schedule a task in this case.

Reply:
Right we could -> will do :)